### PR TITLE
Save binary response with `save_file` function

### LINKED ISSUE #67: Add `save_file` util function
Since many API calls return binary data like getJP2Image and downloadMovie, it would be good to have a function available to quickly save those to disk.

`def save_file(data: bytearray, filename: str)`

### PR DESCRIPTION
**Pull request by akash5100**
* closes #67

### CODE REVIEW

File: hvpy/utils.py [hunk 1]
```diff
@@ -127,3 +129,26 @@ def create_events(events: List[Union[EventType, str, tuple]]) -> str:
             raise ValueError(f"{event} is not a EventType or str or two-length tuple")
     # Strips the final comma
     return constructed_events[:-1]
+
+
+def save_file(data: bytearray, filename: str, overwrite: Optional[bool] = False) -> None:
+    """
```
Comment (nabobalis) [inline hvpy/utils.py:135]: Optional is only it's allowed to be None.

File: hvpy/utils.py [hunk 2]
```diff
@@ -127,3 +129,26 @@ def create_events(events: List[Union[EventType, str, tuple]]) -> str:
...
+        Whether to overwrite the file if it already exists.
+        Default is `False`, optional.
+    """
+    if Path(filename).exists():
+        if overwrite:
+            Path(filename).write_bytes(data)
+        else:
+            raise ValueError(f"{filename} already exists, use overwrite=True to overwrite")
+    else:
+        Path(filename).write_bytes(data)
```
Comment (nabobalis) [inline hvpy/utils.py:154]: This should be condensed, you should do the logic check once, raise the error or then write the file.

File: hvpy/utils.py [hunk 3]
```diff
@@ -127,3 +129,25 @@ def create_events(events: List[Union[EventType, str, tuple]]) -> str:
             raise ValueError(f"{event} is not a EventType or str or two-length tuple")
     # Strips the final comma
     return constructed_events[:-1]
+
+
+def save_file(data: bytearray, filename: str, overwrite: Optional[bool] = None) -> None:
```
Comment (nabobalis) [inline hvpy/utils.py:134]: Overwrite should be false by default. It isn't optional aka none

Comment (nabobalis) [inline hvpy/utils.py:134]: We should also check that filename works with a Path object

File: hvpy/utils.py [hunk 4]
```diff
@@ -127,3 +129,25 @@ def create_events(events: List[Union[EventType, str, tuple]]) -> str:
...
+    overwrite
+        Whether to overwrite the file if it already exists.
+        Default is `False`, optional.
+    """
+    if Path(filename).exists() and overwrite is None:
+        raise ValueError(f"{filename} already exists. Use overwrite=True to overwrite.")
+    elif Path(filename).exists() and overwrite is True:
+        Path(filename).write_bytes(data)
+    else:
+        Path(filename).write_bytes(data)
```
Comment (nabobalis) [inline hvpy/utils.py:153]: This should be around three lines of code, with one write command.

File: hvpy/utils.py (hunk 3 above)
Comment (akash5100) [reply hvpy/utils.py:134]: We accept `filename` as an `str` and string works with Path, how can we check it?

Comment (nabobalis) [reply hvpy/utils.py:134]: You pass in a Path object in a test and will need to update the type hint as well.

File: hvpy/utils.py [hunk 5]
```diff
@@ -127,3 +129,23 @@ def create_events(events: List[Union[EventType, str, tuple]]) -> str:
...
+    filename
+        The path to save the file to.
+    overwrite
+        Whether to overwrite the file if it already exists.
+        Default is `False`.
+    """
+    if filename.exists() and not overwrite:
+        raise ValueError(f"{filename} already exists. Use overwrite=True to overwrite.")
+    else:
+        filename.write_bytes(data)
```
Comment (nabobalis) [inline hvpy/utils.py:151]:
```suggestion
    if filename.exists() and not overwrite:
        raise ValueError(f"{filename} already exists. Use overwrite=True to overwrite.")
    filename.write_bytes(data)
```

File: hvpy/utils.py [hunk 6]
```diff
@@ -127,3 +129,23 @@ def create_events(events: List[Union[EventType, str, tuple]]) -> str:
             raise ValueError(f"{event} is not a EventType or str or two-length tuple")
     # Strips the final comma
     return constructed_events[:-1]
+
+
+def save_file(data: bytearray, filename: Path, overwrite: bool = False) -> None:
```
Comment (nabobalis) [inline hvpy/utils.py:134]:
```suggestion
def save_file(data: bytearray, filename: Union[Path, str], overwrite: bool = False) -> None:
```
I think you will need to import Union

**Review by nabobalis — APPROVED** [review]

**Review by dgarciabriseno — APPROVED** [review]

**Pull request merged**